### PR TITLE
compiler support for exception throwing

### PR DESCRIPTION
**Pull request by csmuller**
This is only about throwing exceptions not about the try-catch-finally blocks.

### CODE REVIEW

**Comment by gsmachado** [comment]
> This is only about throwing exceptions not about the try-catch-finally blocks.

Clear. 👍 

Short question: how complex is the try-catch-finally blocks structure translation to NeoVM?

File: compiler/src/main/java/io/neow3j/compiler/converters/ObjectsConverter.java [hunk 1]
```diff
@@ -247,9 +315,9 @@ private static AbstractInsnNode convertConstructorCall(TypeInsnNode typeInsn,
     }
 
     // Checks if the given instruction is a call to the given classes constructor (i.e., <init>).
-    private static boolean isCallToCtor(AbstractInsnNode insn, ClassNode owner) {
+    private static boolean isCallToCtor(AbstractInsnNode insn, String ownerInternalName) {
```
Comment (gsmachado) [inline compiler/src/main/java/io/neow3j/compiler/converters/ObjectsConverter.java:318]: This change was necessary just to make the comparison in an easier way, right?

Or are there more reasons? Just trying to understand.

**Comment by csmuller** [comment]
> Short question: how complex is the try-catch-finally blocks structure translation to NeoVM?

That's not a short question ;)
I don't have the full picture yet. Will have it till the end of the day. Then I can tell you more. But it seems possible in the foreseen time.

**Comment by csmuller** [comment]
This pull-request can be merged independent of the try-catch functionality.

**Review by gsmachado — APPROVED** [review]
Good stuff! 👍 

So, just for my understanding:

By reading the `CompilerExceptionsTest` and `ThrowExceptionTest` class, I could get the limitations of the exception/throw mechanism.
Am I wrong, or it means that only `Exception` is allowed to be instantiated and thrown?

File: compiler/src/main/java/io/neow3j/compiler/converters/ObjectsConverter.java (hunk 1 above)
Comment (csmuller) [reply compiler/src/main/java/io/neow3j/compiler/converters/ObjectsConverter.java:318]: Yes, just to make the method usable in more places. Only the name of the owner is needed inside of the method anyway, so why pass the whole ClassNode.

**Comment by csmuller** [comment]
> Am I wrong, or it means that only `Exception` is allowed to be instantiated and thrown?

Indeed, only `Exception` is supported. We can re-evaluate this restriction after implementing the try-catch functionality. Maybe we'll allow the usage of other, e.g. custom exceptions, as well.

**Comment by gsmachado** [comment]
@csmuller great.

It would be nice if we keep these "nice-to-have" features somewhere. Either as an issue or in your ToDos. haha.

Then we could propose further extensions and/or enhancements in the near future. 👍 💯

**Comment by csmuller** [comment]
I have it on my todo list for the moment. We can transfer it to #249 if we find that it makes sense when finishing with #181.

**Comment by gsmachado** [comment]
> I have it on my todo list for the moment. We can transfer it to #249 if we find that it makes sense when finishing with #181.

That's a good approach, thanks. 👍

**Pull request merged**